File: evals/fixtures/hotel-flask-py/app/blueprints/bookings.py
from flask import Blueprint, jsonify, request

from services.booking_service import (
    cancel_booking,
    create_booking,
    get_booking,
    update_booking,
)

bookings_bp = Blueprint("bookings", __name__, url_prefix="/api/bookings")


@bookings_bp.route("", methods=["POST"])
def create():
    try:
        booking = create_booking(request.json)
        return jsonify({"booking": booking}), 201
    except ValueError as e:
        return jsonify({"error": str(e)}), 400


@bookings_bp.route("/<booking_id>", methods=["PUT"])
def update(booking_id):
    try:
        booking = update_booking(booking_id, request.json)
        return jsonify({"booking": booking}), 200
    except LookupError as e:
        return jsonify({"error": str(e)}), 404


@bookings_bp.route("/<booking_id>", methods=["DELETE"])
def cancel(booking_id):
    try:
        booking = cancel_booking(booking_id)
        return jsonify({"booking": booking}), 200
    except LookupError as e:
        return jsonify({"error": str(e)}), 404


@bookings_bp.route("/<booking_id>", methods=["GET"])
def get(booking_id):
    try:
        booking = get_booking(booking_id)
        return jsonify({"booking": booking}), 200
    except LookupError as e:
        return jsonify({"error": str(e)}), 404

File: evals/fixtures/hotel-express-ts/app/src/routes/bookings.ts
import { Router, Request, Response } from "express";
import {
  createBooking,
  updateBooking,
  cancelBooking,
  getBooking,
} from "../services/bookingService";

const router = Router();

router.post("/", (req: Request, res: Response) => {
  try {
    const booking = createBooking(req.body);
    res.status(201).json({ booking });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

router.put("/:id", (req: Request, res: Response) => {
  try {
    const booking = updateBooking(req.params.id, req.body);
    res.status(200).json({ booking });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

router.delete("/:id", (req: Request, res: Response) => {
  try {
    const booking = cancelBooking(req.params.id);
    res.status(200).json({ booking });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

router.get("/:id", (req: Request, res: Response) => {
  try {
    const booking = getBooking(req.params.id);
    res.status(200).json({ booking });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

export default router;
